src/node_sqlite.cc

```diff
@@ -824,14 +826,21 @@ void DatabaseSync::ApplyChangeset(const FunctionCallbackInfo<Value>& args) {
 
       Local<Function> filterFunc = filterValue.As<Function>();
 
-      filterCallback = [env, filterFunc](std::string item) -> bool {
+      context.filterCallback = [env, filterFunc](std::string item) -> bool {
         Local<Value> argv[] = {String::NewFromUtf8(env->isolate(),
-                                                   item.c_str(),
-                                                   NewStringType::kNormal)
-                                   .ToLocalChecked()};
-        Local<Value> result =
-            filterFunc->Call(env->context(), Null(env->isolate()), 1, argv)
-                .ToLocalChecked();
+                                                  item.c_str(),
+                                                  NewStringType::kNormal)
+                                  .ToLocalChecked()};
+        MaybeLocal<Value> maybe_result =
+            filterFunc->Call(env->context(), Null(env->isolate()), 1, argv);
+
+        if (maybe_result.IsEmpty()) {
+          return false;
+        }
```

Comment:
Not technically wrong, but you don't need this because you already handle the `empty` case in the next conditional

```suggestion
```

Reply:
I will still add some logic here, I think this branch is taken when an exception is thrown, correct?

Reply:
Yes, it should be taken when `filterFunc` throws an exception. But the point of the comment here is that the next condition `!maybe_result.ToLocal(&result)` returns `false` in the same case.